### ISSUE TITLE
Setup with textarea doesn't attach event listeners correctly

### ISSUE THREAD

**Issue by bborn** [daviferreira/medium-editor#580]
See #533 : setup function needs to `initElements` before attaching event listeners (`initCommands`), otherwise the listeners get attached to the textarea, not the contenteditable div. This breaks things like the mouseover anchor link preview functionality, image drag/drop, etc.

@nmielnik

**Comment by nmielnik** [comment]
The reason why we're doing things in this order is that we create the Toolbar inside `initElements()`.  The toolbar will look through all the extensions that implement a `.getButton` function and call it so that the extension can return an element that will be placed inside the toolbar.

Extensions assume that `.init()` is called before `.getButton()`, so this is why `initCommands()` (which calls `.init()` on each extension) must be called before `initElements()`.

I believe the solution here is to extract the toolbar initiating code out of `initElements()` into its own `initToolbar()` method.  Once that's done, we can change the `setup()` function to this:

```js
setup: function () {
  ...
  // Call initialization helpers
  initElements.call(this);
  initCommands.call(this);
  initToolbar.call(this); // New method we've created
  ...
}
```

If anyone has a chance to open a pull request for this in the meantime, that'd be great!

**Comment by nmielnik** [comment]
This should be fixed by #583 and released as part of v4.7.0
